### PR TITLE
GEODE-9908: when make durable client queue primary, send marker messa…

### PR DESCRIPTION
**Pull request by jinmeiliao**
…ge directly instead of enqueue the marker

### CODE REVIEW

**Review by boglesby — APPROVED** [review]

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  public static class KeyValueCacheListener extends CacheListenerAdapter<Object, Object> {
+    private static Logger logger = LogService.getLogger();
+    public int updateEventCount;
+    public int markerEventCount;
+
+    @Override
+    public void afterUpdate(EntryEvent event) {
+      logger.info("Jinmei: got event {}", event.getOperation().toString());
```
Comment (pivotal-eshu) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java:151]: Debug logging info should be removed.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      updateEventCount++;
+    }
+
+    public void afterRegionLive(RegionEvent event) {
+      markerEventCount++;
+      logger.info("Jinmei: got event {}", event.getOperation().toString());
```
Comment (pivotal-eshu) [inline geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java:157]: Debug logging info should be removed.

File: geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java [hunk 3]
```diff
@@ -1673,13 +1673,16 @@ MessageDispatcher createMessageDispatcher(String name) {
...
     // Only start or resume the dispatcher if it is Primary
     if (isPrimary) {
       // Add the marker to the queue
-      if (!processedMarker) {
-        EventID eventId = new EventID(_cache.getDistributedSystem());
-        _messageDispatcher.enqueueMarker(new ClientMarkerMessageImpl(eventId));
+      EventID eventId = new EventID(_cache.getDistributedSystem());
+      ClientMarkerMessageImpl clientMarkerMessage = new ClientMarkerMessageImpl(eventId);
+      if (sendMarkerDirectly) {
+        sendMessageDirectly(clientMarkerMessage);
```
Comment (pivotal-eshu) [inline geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java:1683]: Will this cause unforeseen consequences if Dispatcher is not yet started/resumed but send message first? The method later starts dispatcher thread.

Comment (jinmeiliao) [reply geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/CacheClientProxy.java:1683]: I can't see any bad consequence. It's just a marker message that is supposed to go to the client anyway.

File: geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java (hunk 1 above)
Comment (jinmeiliao) [reply geode-core/src/distributedTest/java/org/apache/geode/internal/cache/tier/sockets/SlowDispatcherDUnitTest.java]: done

**Review by kirklund — APPROVED** [review]

**Review by dschneider-pivotal — APPROVED** [review]

**Pull request merged**